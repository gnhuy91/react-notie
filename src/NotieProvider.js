import React, { Component, PropTypes } from 'react';

import notieContextShape from './context-shape';
import Notie from './Notie';

class NotieProvider extends Component {
    static childContextTypes = {
        notie: notieContextShape
    };

    static propTypes = {
        children: PropTypes.node.isRequired
    }

    /* istanbul ignore next */
    getChildContext() {
        return {
            notie: {
                alert: (...props) => this.notie.alert(...props),
                confirm: (...props) => this.notie.confirm(...props),
                success: (...props) => this.notie.success(...props),
                warn: (...props) => this.notie.warn(...props),
                error: (...props) => this.notie.error(...props),
                info: (...props) => this.notie.info(...props)
            }
        };
    }

    notieRef = (c) => {
        this.notie = c;
    }

    render() {
        return (
            <div>
                {this.props.children}
                <Notie ref={this.notieRef} />
            </div>
        );
    }
}

export default NotieProvider;
