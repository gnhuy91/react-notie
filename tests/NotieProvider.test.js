import React from 'react';
import { shallow, mount } from 'enzyme';

import NotieProvider from '../src/NotieProvider';
import contextShape from '../src/context-shape';

describe('<NotieProvider />', () => {
    let component;

    beforeEach(() => {
        component = shallow(
            <NotieProvider>
                <div>some children</div>
            </NotieProvider>
        );
    });

    afterEach(() => {
        component.unmount();
    });

    test('renders correctly with children', () => {
        expect(component).toMatchSnapshot();
    });

    test('has notieRef', () => {
        expect(component.instance().notieRef).toBeDefined();
    });

    describe('context', () => {
        let rootContext;
        let componentWithContext;
        function ContextChecker(props, context) {
            rootContext = context;
            return null;
        }

        ContextChecker.contextTypes = {
            notie: contextShape
        };

        beforeEach(() => {
            componentWithContext = mount(
                <NotieProvider>
                    <ContextChecker />
                </NotieProvider>
            );
        });

        afterEach(() => {
            componentWithContext.unmount();
            rootContext = undefined;
        });

        test('alert method', () => {
            expect(rootContext.notie).toHaveProperty('alert');
            expect(typeof rootContext.notie.alert).toBe('function');
        });

        test('confirm method', () => {
            expect(rootContext.notie).toHaveProperty('confirm');
            expect(typeof rootContext.notie.confirm).toBe('function');
        });

        test('success method', () => {
            expect(rootContext.notie).toHaveProperty('success');
            expect(typeof rootContext.notie.success).toBe('function');
        });

        test('warn method', () => {
            expect(rootContext.notie).toHaveProperty('warn');
            expect(typeof rootContext.notie.warn).toBe('function');
        });

        test('error method', () => {
            expect(rootContext.notie).toHaveProperty('error');
            expect(typeof rootContext.notie.error).toBe('function');
        });

        test('info methods', () => {
            expect(rootContext.notie).toHaveProperty('info');
            expect(typeof rootContext.notie.info).toBe('function');
        });
    });
});
