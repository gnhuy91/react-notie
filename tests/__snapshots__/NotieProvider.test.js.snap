// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotieProvider /> renders correctly with children 1`] = `
<div>
  <div>
    some children
  </div>
  <Notie />
</div>
`;
